[Feat] #44 - AWS SNS 세팅

yangsubinn opened the pull request:
### 🛠 작업 내용
<!-- 작업한 내용을 적어주세요. -->
- aws-sdk-ios-spm 추가
- AWS SNS 관련 appdelegate 세팅
- UserDefault에 token 저장

### 🚨 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->
AWS SNS .. 넣어두긴 했는데, 사실 이게 맞는지 잘 모르겠어요 .. 자료가 많이 없기도 하고 나와도 다 몇년 전 자료들이라 .. 서버에서 푸시 쏴줘봐야 알것 같아요 .. 🥺


### 📸 스크린샷
|기능|스크린샷|
|:--:|:--:|
|GIF|파일첨부|

### 💡 관련 이슈
<!-- 관련있는 이슈 번호(#000)를 적어주세요. -->
closed #44

yangsubinn committed:
[Setting] #44 - AWS SDK iOS SPM 추가

yangsubinn committed:
[Chore] #44 - userDefaultKeyList에서 UserDefaultWrapper 사용하여 선언

yangsubinn committed:
[Feat] #44 - AWS SNS 관련 세팅

lsj8706 reviewed (approved):
수고하셨습니다~~! 나중에 서버에서 푸시알림 보내주면 테스트 해봐요~!!

L-j-h-c commented on SOPT-iOS/Projects/SOPT-iOS/Sources/Application/AppDelegate.swift:
```diff
@@ -36,6 +49,28 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
+        
+        // create platform endpoint (= device endpoint ARN)
+        let sns = AWSSNS.default()
+        let request = AWSSNSCreatePlatformEndpointInput()
+        request?.token = token
+        request?.platformApplicationArn = SNSPlatformApplicationArn
+        sns.createPlatformEndpoint(request!).continueWith(executor: AWSExecutor.mainThread(), block: { (task: AWSTask!) -> AnyObject? in
+            if task.error != nil {
+                print("Error: \(String(describing: task.error))")
+            } else {
+                let createEndpointResponse = task.result! as AWSSNSCreateEndpointResponse
+                
+                if let endpointArnForSNS = createEndpointResponse.endpointArn {
+                    print("endpointArn: \(endpointArnForSNS)")
+                    UserDefaultKeyList.Auth.endpointArnForSNS = endpointArnForSNS
+                }
+            }
+            
+            return nil
+        })
```
개인적으로 요런 기능 담당하는 푸시알림 매니저 클래스가 있는 것도 좋을 것 같아요! 서비스나 레포지토리가 있어도 좋구... 그냥 의견입니다🙃

yangsubinn replied:
조금 구성이 많아서 분리해두면 좋긴 하겠네욤 .. 일단은 .. 제대로 작동할지도 아직 모르겠어서 두고 잘 작동된다고 하면 분리해보도록 할게여 !

L-j-h-c reviewed (approved):
감사합니다~~ 저도 공부해보겠슴당

devxsby reviewed (approved):
어렵당. . .
저도 함 다시 알아보면서 코드 봐볼게여
고생하셨수빙수 💗

yangsubinn committed:
[Feat] #44 - AWS SNS ARN 키 추가

yangsubinn committed:
[Merge] develop pull
Merge branch 'develop' of https://github.com/sopt-makers/SOPT-iOS into feature/#44-AWSSNS

devxsby reviewed (commented):
감사합니다 🚀
